source/megengine.js

```diff
@@ -3,30 +3,96 @@
 
 var megengine = megengine || {};
 var base = base || require('./base');
+var zip = zip || require('./zip');
 
 megengine.ModelFactory = class {
 
     match(context) {
+        megengine.modelType = '';
+        const identifier = context.identifier;
+        const extension = identifier.split('.').pop().toLowerCase();
+        const stream = context.stream;
+        if (extension === 'mge' && stream.length >= 8) {
+            const buffer = stream.peek(Math.min(32, stream.length));
+            if (buffer.length >= 12) {
+                const magic = String.fromCharCode(...buffer.slice(8, 12));
```

Comment:
Use `flatbuffers.BinaryReader` to check `file_identifier`. See `./source/mslite.js` for example

Reply:
done

Reply:
What is the purpose of the size stored in the initial 4 bytes and should there be a check it is a valid size?

Reply:
The initial 4 bytes store the size of the model. The main purpose is to facilitate the user to know the size of the model when loading. 
Yes, I added a check.

Reply:
Running in the browser using `npm run server -- -b` shows this error:

```
TypeError stream.peek(4).readUInt32LE is not a function. (In 'stream.peek(4).readUInt32LE(0)', 'stream.peek(4).readUInt32LE' is undefined) in 'mobilenetv3-large.mge'.
```

```
const buffer = stream.peek(4)
const reader = new base.BinaryReader(buffer);
const size = reader.uint32();
if (size === stream.length) {
    ...
}

Reply:
done